streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/core/service/impl/ApplicationServiceImpl.java

```diff
@@ -1620,11 +1636,16 @@ private Map<String, Object> getProperties(Application application) {
       properties.put(RestOptions.ADDRESS.key(), activeAddress.getHost());
       properties.put(RestOptions.PORT.key(), activeAddress.getPort());
     } else if (ExecutionMode.isYarnMode(application.getExecutionModeEnum())) {
+
       String yarnQueue =
           (String) application.getHotParamsMap().get(ConfigConst.KEY_YARN_APP_QUEUE());
-      if (yarnQueue != null) {
-        properties.put(ConfigConst.KEY_YARN_APP_QUEUE(), yarnQueue);
-      }
+      String yarnLabelExpr =
+          (String) application.getHotParamsMap().get(ConfigConst.KEY_YARN_APP_NODE_LABEL());
+      Optional.ofNullable(yarnQueue)
+          .ifPresent(yq -> properties.put(ConfigConst.KEY_YARN_APP_QUEUE(), yq));
+      Optional.ofNullable(yarnLabelExpr)
+          .ifPresent(yLabel -> properties.put(ConfigConst.KEY_YARN_APP_NODE_LABEL(), yLabel));
+
```

Comment:
The code block would be enter when the mode is yarn-mode, in the truth, the queue-label value would be set when the following condition is true:
```
private boolean needFillYarnQueueLabel(ExecutionMode mode) {
    return ExecutionMode.YARN_PER_JOB.equals(mode) || ExecutionMode.YARN_APPLICATION.equals(mode);
  }
```

so, the current code segment is correct for the context of the application lineage, just not exact enough.

I'll give the change in the next commit.

Reply:
So, please let me know what's your opinions.
thx~!